core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -19,34 +19,45 @@
 
 package org.elasticsearch.index.query;
 
+import com.sun.jna.Memory;
+
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.geo.GeoHashUtils;
 import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.fielddata.IndexGeoPointFieldData;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.geo.GeoPointFieldMapper;
+import org.elasticsearch.index.search.geo.InMemoryGeoBoundingBoxQuery;
+import org.elasticsearch.index.search.geo.IndexedGeoBoundingBoxQuery;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class GeoBoundingBoxQueryBuilder extends AbstractQueryBuilder<GeoBoundingBoxQueryBuilder> {
-
     public static final String NAME = "geo_bbox";
 
-    public static final String TOP_LEFT = GeoBoundingBoxQueryParser.TOP_LEFT;
-    public static final String BOTTOM_RIGHT = GeoBoundingBoxQueryParser.BOTTOM_RIGHT;
-
-    private static final int TOP = 0;
-    private static final int LEFT = 1;
-    private static final int BOTTOM = 2;
-    private static final int RIGHT = 3;
+    public static final double MAX_LAT = 90.0;
+    public static final double MIN_LAT = -90.0;
+    public static final double MAX_LON = 180.0;
+    public static final double MIN_LON = -180.0;
 
-    private final String name;
+    private GeoPoint topLeft = null;
+    private GeoPoint bottomRight = null;
 
-    private double[] box = {Double.NaN, Double.NaN, Double.NaN, Double.NaN};
-
-    private String type;
+    private Type type;
+    private String fieldName;
 
     static final GeoBoundingBoxQueryBuilder PROTOTYPE = new GeoBoundingBoxQueryBuilder(null);
 
-    public GeoBoundingBoxQueryBuilder(String name) {
-        this.name = name;
+    public GeoBoundingBoxQueryBuilder(String fieldName) {
+        this.fieldName = fieldName;
```

Comment:
Noticed only after pushing - rejecting null as fieldName here doesn't actually work unless we change the initialisation of PROTOTYPE to a non-empty string which IIRC we didn't want to do for the SpanTermQueryBuilder...